File: aoc/2021/1.py
import numpy as np

with open("input.txt", "r") as f:
    lines = f.readlines()


def part_1_2(depths, window_size):
    diff = depths[window_size:] - depths[:-window_size]
    return sum([1 for x in diff if x > 0])


if __name__ == "__main__":
    depths = np.array([int(line.strip()) for line in lines])
    print(part_1_2(depths))

File: aoc/2021/1.cc
#include <bits/stdc++.h>
#include <fstream>
#include <iostream>
#include <string>

using namespace std;

int solve(const vector<int>& depths, int window_size) {
    int res = 0;

    for (int i = 0; i + window_size < (int) depths.size(); i++) {
        if (depths[i + window_size] - depths[i] > 0) res += 1;
    }

    return res;
}

int main (int argc, char *argv[])
{
    string input_file = "input.txt";
    ifstream fin(input_file, ifstream::in);
    vector<int> depths;

    for(string line; getline(fin, line); ) {
        int depth = stoi(line);
        depths.push_back(depth);
    }

    cout << solve(depths, 3);
    
    return 0;
}
